{{cookiecutter.repo_name}}/{{cookiecutter.package_name}}/utils/video.py

```diff
@@ -0,0 +1,99 @@
+import urllib
+
+import requests
+from bs4 import BeautifulSoup
+
+
+def get_oembed_info_url(url):
+    '''A helper to get the oEmbed information URL for a video.
+
+    This special-cases YouTube videos because bot requests to video pages
+    frequently result in the IP being captcha'd out, which breaks the
+    auto-discovery mechanism. Fortunately, these IP bans do not seem to apply
+    to requests to `/oembed`.
+    '''
+    youtube_domains = ['youtu.be', 'www.youtube.com', 'youtube.com', 'm.youtube.com']
+
+    if urllib.parse.urlparse(url).netloc in youtube_domains:
+        return 'https://www.youtube.com/oembed?format=json&{}'.format(
+            urllib.parse.urlencode({'url': url})
+        )
+
+    # The "not YouTube" case - this should reliably handle everything
+    # including Vimeo.
+    try:
+        req = requests.get(url)
+        text = req.text
+        soup = BeautifulSoup(text, 'html.parser')
+    except:  # pylint:disable=bare-except
+        # Either requests failed, or it looked nothing like HTML.
+        return
+
+    # Video providers that support oEmbed will have something that looks like
+    # this:
+    # <link rel='alternate' type='application/json+oembed' href='...'>
+    # Where the contents of 'href' tell us where to go to get JSON
+    # for an embed code.
+    try:
+        rel_tag = soup.find(attrs={
+            'type': 'application/json+oembed'
+        })
+        assert rel_tag.get('href')
+    except:  # pylint:disable=bare-except
+        # This can probably happen if a video is private or deleted.
+        return
+
+    # Now, let's grab the JSON
+    return rel_tag.get('href')
+
+
+def get_video_info(url):
+    '''Returns video information for a given URL. Returns a dict in this form:
```

Comment:
Does this doc string need updating now that it's returning a json obj rather than a dictionary

Reply:
Nope, this will be the parsed JSON, which will be a dict in this case.